### ISSUE TITLE
error with two digit major version (>=10)

### ISSUE THREAD

**Issue by ddrogahn** [NuGet/Home#6305]
NuGet product used: VS UI
NuGet version: 4.4.1.4633
VS version: 15.4.4
OS version: Win10 1709 16299.64

Repro:
1. Install a package with version > 10
eg Newtonsoft.Json 10.0.3

2. Try to install a package that requires the previous package with a version constraint
eg Microsoft.AspNet.WebApi.Client 5.2.3 constraint: Newtonsoft.Json (>= 6.0.4)

3. Dependencies do resolve if first package is manually downgraded to a version less than 10

### Log
Attempting to gather dependency information for multiple packages with respect to project 'Microsoft.Bot.Builder.RealTimeMediaCalling', targeting '.NETFramework,Version=v4.6'
Gathering dependency information took 652.92 ms
Attempting to resolve dependencies for multiple packages.
Unable to resolve dependencies. 'Newtonsoft.Json 10.0.3' is not compatible with
 'Microsoft.AspNet.WebApi.Client 5.2.3 constraint: Newtonsoft.Json (>= 6.0.4)',
 'Microsoft.Bot.Builder 3.12.2.4 constraint: Newtonsoft.Json (>= 9.0.1)',
 'Microsoft.Bot.Builder.Calling 3.12.2.4 constraint: Newtonsoft.Json (>= 8.0.3)',
 'Microsoft.Bot.Connector 3.12.2.4 constraint: Newtonsoft.Json (>= 9.0.1)',
 'Microsoft.Rest.ClientRuntime 2.3.10 constraint: Newtonsoft.Json (>= 6.0.8)'.

**Comment by emgarten** [comment]
This is typically caused by a feed that treats versions as strings. Older versions of NuGet.Server did this.

What feeds are you using? Are you able to repro this after copying the package to a local folder and using that as the source?

**Comment by ddrogahn** [comment]
Just the default https://api.nuget.org/v3/index.json

I did later get the packages to install by first downgrading json to 9, then installing packages, and upgrading to json 10.

Today I got NuGet 4.5.0 with VStudio 15.5.1 -- and I can't reproduce it now.

It doesn't seem like just restarting fixed it.  I did have a pretty fresh instance of VS when I saw this yesterday.  And have seen this problem before -- months ago, on a different (Win7) machine.
(Newtonsoft.json v10 is 9 months old)

In any case, I guess that's a potential workaround and a potential fix to a unconfirmed problem.
